src/asynk/async_queue.rs

```diff
@@ -552,19 +483,30 @@ where
         transaction: &mut Transaction<'_>,
         metadata: serde_json::Value,
         task_type: &str,
+        scheduled_at: DateTime<Utc>,
     ) -> Result<Task, AsyncQueueError> {
-        match Self::find_task_by_metadata_query(transaction, &metadata).await {
+        match Self::find_task_by_uniq_hash_query(transaction, &metadata).await {
             Some(task) => Ok(task),
-            None => Self::insert_task_query(transaction, metadata, task_type).await,
+            None => {
+                Self::insert_task_uniq_query(transaction, metadata, task_type, scheduled_at).await
+            }
         }
     }
 
-    async fn find_task_by_metadata_query(
+    async fn find_task_by_uniq_hash_query(
         transaction: &mut Transaction<'_>,
         metadata: &serde_json::Value,
     ) -> Option<Task> {
+        let mut hasher = Sha256::new();
```

Comment:
hash calculation can be moved into a separate function

Reply:
[comments suggestions](https://github.com/ayrat555/fang/pull/71/commits/4eb86a2e2a6a2e33153736332de9191e4a162d56)